### PR TITLE
Community 2.0 doc updates

### PR DESCRIPTION
**Pull request by ganesaar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,10 +3,9 @@
 #### Product version: 1.2.0
```
Comment (suresh-ls) [inline README.md]: ->  2.0.0

File: README.md [hunk 2]
```diff
@@ -59,19 +58,23 @@ NOTE: All the ML nodes must use the same ML platform either Keras (based on Tens
...
-    **Caution**: Users are recommended not to save their model related artifacts under this folder, as future version upgrade of Swarm Learning would delete these folders.
-  - [Upgrading from earlier evaluation versions](/docs/Install/Versioning_and_upgrade.md)
-  - [Download and setup Swarm Learning](/docs/Install/HPE_Swarm_Learning_installation.md) using the Web UI installer 
-  - Execute [MNIST example](/examples/mnist/MNIST.md) 
-  - [Frequently Asked Questions](/docs/User/Frequently_asked_questions.md)
-  - [Troubleshooting](/docs/User/Troubleshooting.md)
+  1. [Prerequisites](/docs/Install/Prerequisites.md) for Swarm Learning
+  2. [Upgrading from earlier evaluation versions](/docs/Install/Versioning_and_upgrade.md)
+  3. [Download and setup Swarm Learning](/docs/Install/HPE_Swarm_Learning_installation.md) using the Web UI installer
```
Comment (suresh-ls) [inline README.md]: All Web UI -> SLM-UI

File: docs/Install/Creating_a_Contract.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# <a name="GUID-5EBE2D57-5A69-4C12-996B-AB854DF657C2"/> Creating a Contract
+
+1.  In the **MyProjects**, click the **Project name** link.
+
+2.  In the **Contracts** tab, click **Create new contract**.
+
+    If you are running concurrent Swarm training, you need to create multiple contracts. If you are running a single training, the default contract would be good enough and this step is optional.
```
Comment (suresh-ls) [inline docs/Install/Creating_a_Contract.md:7]: Add a note here "This feature is available for enterprise customers ONLY"

File: README.md [hunk 4]
```diff
@@ -80,6 +83,8 @@ NOTE: All the ML nodes must use the same ML platform either Keras (based on Tens
   - [Running Swarm Learning Components](/docs/Install/Running_Swarm_Learning.md)
   - [Using SWCI](/docs/User/Swarm_Learning_Command_Interface.md)
   - [Using SWOP](/docs/User/Swarm_Operator_node_(SWOP).md)
+  - [Examples](/examples/README.md)
+  - [Swarm Learning Log Collection](/docs/User/Swarm_Log_Collector.md)
```
Comment (suresh-ls) [inline README.md:88]: After examples:

1. We will add a new item "Running Swarm learning examples using SLM-UI and point to the content "Page 15 to 24" of install guide

2. We will move "Running Swarm Learning Components" from above and call it as "Running Swarm Learning using CLI"

File: docs/Install/Creating_Task_Runners.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# <a name="GUID-CAD82E68-9A3A-465E-9DC2-06BBDCD1DC55"/> Creating Task Runners
+
+1.  In the **MyProjects**, click the **Project name** link.
+
+2.  In the **Task Runners** tab, click **Create new task runner**.
+
+    If you are running concurrent Swarm training, you need to create multiple task runners. If you are running a single training, the default task runner would be good enough and this step is optional.
```
Comment (suresh-ls) [inline docs/Install/Creating_Task_Runners.md:7]: Add a note here "This feature is available for enterprise customers ONLY"

File: docs/Install/HPE_Swarm_Learning_installation.md [hunk 6]
```diff
@@ -2,6 +2,6 @@
 
 1.   [Download the License Server](Download_the_License_Server.md) 
 2.   [Install the License Server and download Swarm Learning](Install_the_License_Server.md) 
-3.   [Install HPE Swarm Learning](Install_HPE_Swarm_Learning.md) 
+3.   [Installing HPE Swarm Learning Management UI](Installing_HPE_Swarm_Learning_Management_UI.md)
```
Comment (suresh-ls) [inline docs/Install/HPE_Swarm_Learning_installation.md]: 1. Please remove point 3.

Add this for point 3:

Installing Swarm Learning is a two-step process.

a.  Using SLM-UI Installer, you can install the SLM-UI on one host ( and point to pages 11 to 14 of install guide "Running SLM_UI installer.md" )
b.  Using SLM-UI, you can install Swarm Learning in multiple hosts.  ( and point to "Adding a Swarm Host in SLM-UI.md"

2. Also the content of point 2 link needs a change in version no. .  "HPE-SWARM-CMT 1.2.0"  to be changed to "HPE-SWARM-CMT2.0.0"

File: docs/Install/IP_address_of_host_systems.md [hunk 7]
```diff
@@ -1,4 +1,9 @@
 # <a name="GUID-1D2971AF-D537-4521-ACAA-D910F50C43E0"/> IP address of host systems
 
-By default, Swarm Learning uses the host network to connect and communicate with its peers. In this case, the IP addresses represent the IP addresses or a fully qualified domain name \(FQDN\) of the host systems on which the containers are run. Swarm Learning can be configured to use a Docker bridge network. In this case, the IP addresses represent the IP addresses or FQDN of the containers themselves.
+The `--host-ip` and `slhostip` IP addresses in the run scripts and the SWOP profile are the IP addresses of the host machine, where the respective containers are running on the host machine. Based on access, user can even use the FQDN of the host system.
```
Comment (suresh-ls) [inline docs/Install/IP_address_of_host_systems.md]: Use this new content and remove the old content:

"By default, Swarm Learning framework uses a Docker bridge network. For improved isolation, users can even use a user-defined (custom) bridge network.

"The --host-ip and _slhostip_ IP addresses in the run scripts and the SWOP profile are the "IP addresses" of the host machine, where the respective containers are running. If user-defined docker bridge network is used, one can even use the FQDN of the host system.

While using the user-defined bridge network, the options "--ip" for run-scripts and "ip" field of _slnetworkopts_ in SWOP profile are
the IP addresses of the containers ( not host IP). This case is specific to the reverse proxy examples or scenarios where user wants to use fixed IP address for the containers."

File: docs/Install/Installing_HPE_Swarm_Learning_Management_UI.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# <a name="GUID-60017971-B0A9-4119-AEAF-A21594EE5C1E"/> Installing HPE Swarm Learning Management UI \(SLM-UI\)
+
+Installing Swarm Learning is a two-step process.
+
+1.  Using SLM-UI Installer, you can install the SLM-UI on one host.
+2.  Using SLM-UI, you can install SL in multiple hosts and run the examples.
+
+## Procedure
```
Comment (suresh-ls) [inline docs/Install/Installing_HPE_Swarm_Learning_Management_UI.md]: This file can be deleted..  This way of pulling images is ap[applicable only for enterprise and not community users.

File: docs/Install/Running_SLM-UI_Installer.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    This concludes the installation of SLM-UI installer. The SLM-UI will be launched in a separate browser window.
+
+    ![](GUID-E00B84BB-E871-4A2E-A5E4-0D08B070DA57-high.png)
+
+    User can login into the SLM-UI using default account \(Username: admin; Password: admin\).
+
+    After logging in to the SLM-UI, user can run the examples. User can also create the custom account.
+
+
+**Parent topic:**[Installing HPE Swarm Learning Management UI \(SLM-UI\)](GUID-60017971-B0A9-4119-AEAF-A21594EE5C1E.md)
```
Comment (suresh-ls) [inline docs/Install/Running_SLM-UI_Installer.md]: parent link is broken..  Please do the URL broken check for all links

File: docs/Install/Running_Swarm_Learning_examples_using_SLM-UI.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
+# <a name="GUID-A2B92980-7281-4B0A-989F-33097B7C96A5/"> Running Swarm Learning examples using SLM-UI
+
+**TIP:**
+
+Before you start using the SLM-UI, HPE recommends you to read the *HPE Swarm Learning User Guide* to understand about the architecture of Swarm Learning, how these nodes work, how model training happens, and Swarm Learning Concepts section.
```
Comment (suresh-ls) [inline docs/Install/Running_Swarm_Learning_examples_using_SLM-UI.md]: Reference to user guide to be dropped and instead use URL to main landing page#Documentation section

File: docs/Install/Running_Swarm_Learning_examples_using_SLM-UI.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+6.  [Creating a Contract](Creating_a_Contract.md)
+
+7.  [Executing a Task](Executing_a_Task.md)
+
+8.  [Managing the Global Settings](Managing_the_Global_Settings.md)
+
+
+
+**Parent topic:**[Installing HPE Swarm Learning Management UI \(SLM-UI\)](GUID-60017971-B0A9-4119-AEAF-A21594EE5C1E.md)
```
Comment (suresh-ls) [inline docs/Install/Running_Swarm_Learning_examples_using_SLM-UI.md]: parent link is broken.. Please do the URL broken check for all links

File: docs/Install/Starting_SLM-UI_manually.md [hunk 12]
```diff
@@ -0,0 +1,32 @@
+# Starting SLM-UI manually 
+
+SLM-UI containers are automatically started by the SLM-UI installer. But, if it is not running on the host, users need to manually start it using the following commands.
```
Comment (suresh-ls) [inline docs/Install/Starting_SLM-UI_manually.md:3]: Is there a reference to this new file ?

We can include this as a point in  "Running Swarm Learning examples using SLM-UI", right at the start.

File: docs/Install/Starting_SLM-UI_manually.md [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+1.  Initially, run `<swarm-learning>/slm-ui/scripts/run-postgres -pw" supersecretpassword"` \(default DB password, unless you have changed the default password of the database using external tools like pgAdmin\).
```
Comment (suresh-ls) [inline docs/Install/Starting_SLM-UI_manually.md:5]: This comment is Not for this file.  

Can we rename "Running_Swarm_Learning.md".  as "Running_Swarm_Learning_using_CLI.md".  There are some references to user guide here, we need to drop it.   Do a search for "user guide" "install guide" and fix them appropriately.

File: docs/SWOP-profile-schema.yaml [hunk 14]
```diff
@@ -211,37 +253,99 @@ properties  :
                                 items       : 
                                     type        : object
                                     uniqueItems : true
+
+                      # In our practice, we are using labels only for the SPIFFE selector not envvars.
+                      sllabels:
+                          oneOf:
+                            - const: ~
+                            - type: object
```
Comment (suresh-ls) [inline docs/SWOP-profile-schema.yaml]: This is old file.    We need to take content from this file 

https://github.hpe.com/hpe/swarm-learning-docs/blob/master/docs/SWOP-profile-schema.yaml

File: docs/User/Frequently_asked_questions.md [hunk 15]
```diff
@@ -139,9 +161,11 @@ For configuring the license server API port, see *AutoPass License Server User G
 
 ## <a name="SECTION_HH5_KNP_NSB"/> What is the IP address used in the run scripts?
 
-By default, Swarm Learning framework uses the host network to connect and communicate with its peers. In this case, the IP addresses represent the IP addresses or FQDN of the host systems on which the containers are run.
+The `--host-ip` and `slhostip` IP addresses in the run scripts and the SWOP profile are the IP addresses of the host machine, where the respective containers are running on the host machine. Based on access, user can even use the FQDN of the host system.
```
Comment (suresh-ls) [inline docs/User/Frequently_asked_questions.md]: Replace this answer using the new content of "docs/Install/IP_address_of_host_systems.md"

File: docs/SWOP-task-schema.yaml [hunk 16]
```diff
@@ -1,6 +1,6 @@
 ---
 ######################################################################
-## (C)Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+## (C)Copyright 2021,2022 Hewlett Packard Enterprise Development LP
```
Comment (suresh-ls) [inline docs/SWOP-task-schema.yaml]: This file is old. Please take content from https://github.hpe.com/hpe/swarm-learning-docs/blob/master/docs/SWOP-task-schema.yaml

File: docs/User/SWCI_APIs.md [hunk 17]
```diff
@@ -6,14 +6,14 @@ The following API methods help to invoke SWCI operations programmatically. It is
 NOTE:Arguments that do not have default values are mandatory.
 
 </blockquote>
-The python3 program needs to import the SWCI class, and then use the below APIs.
+The python3 program needs to import the SWCI class first, and then use the below APIs.
```
Comment (suresh-ls) [inline docs/User/SWCI_APIs.md:11]: There is one more note that is missing. Look at page 26 of User guide.

File: docs/User/SWCI_APIs.md [hunk 18]
```diff
@@ -58,19 +58,19 @@ The python3 program needs to import the SWCI class, and then use the below APIs.
...
 
-swciSrvName = 'SWCI Server Name or IP'
+swciServerName = 'SWCI Server Name or IP'
 snServerName = 'SN Server Name or IP'
 
 # Connect to the SWCI via SWCI_WEB_PORT
-s = sw.Swci(swciSrvName,port=30306) #30306 is the default port
+s = sw.Swci(swciServerName,port=30306) #30306 is the default port
 # Connect to SN and create context
 print(s.createContext('testContext', snServerName))
```
Comment (suresh-ls) [inline docs/User/SWCI_APIs.md]: This change is missed out  "ip=" below

#Connect to SN and create context
print(s.createContext('testContext', ip=snServerName))

File: lib/src/README.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+
+## Build 
+Build depends upon python `build` module so make sure its latest version. Only python 3.x version is supported.
+
+`pip install -U build`
+
+Steps to build client library:
+
+1. Set `SWARM_VER` environment variable to corresponding swarm version
+2. Run python -m build command from current directory
```
Comment (htjain) [inline lib/src/README.md]: `current directory` --> `path of src dir`

File: docs/Install/Creating_a_Contract.md (hunk 3 above)
Comment (ganesaar) [reply docs/Install/Creating_a_Contract.md:7]: Updated

File: docs/Install/Creating_Task_Runners.md (hunk 5 above)
Comment (ganesaar) [reply docs/Install/Creating_Task_Runners.md:7]: Updated

File: docs/User/Troubleshooting.md [hunk 20]
```diff
@@ -26,41 +38,39 @@ Error code: 6002, as shown in the following screenshot occurs when Swarm Learnin
 3.  Verify if the Swarm licenses are installed using APLS web management console. For more information, see APLS User Guide.
 
 
-## Installation of HPE Swarm Learning on air-gaped systems or if the Web UI Installer runs into any issue and not able to install
+## 2. Installation of HPE Swarm Learning on air-gaped systems or if the Web UI Installer runs into any issue and not able to install
```
Comment (suresh-ls) [inline docs/User/Troubleshooting.md]: Old content.

Need to pick this from latest 2.0.0 Release notes.pdf ( Page 5)

Comment (suresh-ls) [reply docs/User/Troubleshooting.md]: "hpe_eval"  path in the docker pull command has changed to "hpe"   Search for this string in the documentation.   We should not have it.

File: docs/Install/IP_address_of_host_systems.md (hunk 7 above)
Comment (ganesaar) [reply docs/Install/IP_address_of_host_systems.md]: Resolved

File: docs/User/Troubleshooting.md [hunk 21]
```diff
@@ -26,41 +38,39 @@ Error code: 6002, as shown in the following screenshot occurs when Swarm Learnin
...
     ```
 The suggested workaround is to start with mapping 4 SL to 1 SN. Then after, slowly scale no of SLs to SN
 
-## SWCI waits for task-runner indefinitely even after task completed or failed
+## 4. SWCI waits for task-runner indefinitely even after task completed or failed
 
 User to ensure no failure in ML code before Swarm training starts. Check using `SWARM_LOOPBACK ENV` and ensure, user coderuns fine and local training completes successfully.
 
-# Generic troubleshooting tips
+## 5. Error while docker pull Swarm Learning images: 'could not rotate trust to a new trusted root'
```
Comment (suresh-ls) [inline docs/User/Troubleshooting.md:75]: This is missed out in enterprise. Please make a note of it in PDF for 2.1 release

File: docs/Install/Installing_HPE_Swarm_Learning_Management_UI.md (hunk 8 above)
Comment (ganesaar) [reply docs/Install/Installing_HPE_Swarm_Learning_Management_UI.md]: Resolved

File: docs/User/Troubleshooting.md (hunk 21 above)
Comment (suresh-ls) [inline docs/User/Troubleshooting.md:75]: New issue. from latest user guide (troubleshooting) to be added.

"Unable to contact API-Server"

File: examples/contrib/breakhis/model/breakhis.py [hunk 22]
```diff
@@ -1,3 +1,18 @@
+############################################################################
```
Comment (suresh-ls) [inline examples/contrib/breakhis/model/breakhis.py]: Radha:

Do we really need this ?  As this contrib and not HPE developed ?  What's the legal implication ?

File: docs/Install/Running_Swarm_Learning_examples_using_SLM-UI.md (hunk 11 above)
Comment (ganesaar) [reply docs/Install/Running_Swarm_Learning_examples_using_SLM-UI.md]: Resolved

File: docs/User/Swarm_Learning_Command_Interface.md [hunk 23]
```diff
@@ -66,8 +66,8 @@ A SWCI tool can control operations on multiple Swarm Learning framework instance
...
+|`ASSIGN TASK <taskName : string> TO <trName : string> WITH <peersNeeded> PEERS`|Assigns a task to a taskrunner instance and specify the minimum peer count for declaring the result. This triggers SWOPs to execute this task<br>-   `taskName`: Specifies the unique name of task.<br>-   `trName`: Specifies the unique name of Taskrunner. SWOP’s listening on this Taskrunner participates and executes this task.<br>-   `WITH <peersNeeded> PEERS`:<br>`peersNeeded` is a nonzero positive integer specifying the minimum number of peers required to complete this task.<br>  <br> Meaningful value of minimum peers is dependent on the task type. For RUN\_SWARM task, it is the number of SL and ML node pairs. For all other TASK types, it is the number of SWOP nodes.<br>-  For RUN_SWARM task type, the actual numbers of SL/ML peers started could be equal or greater than `peersNeeded`, depending on the number of the SL nodes defined in the SWOP profiles.<br>- For other TASK types, the number of SWOPs participating would be equal to `peersNeeded`.<br>|
```
Comment (suresh-ls) [inline docs/User/Swarm_Learning_Command_Interface.md]: For this and the next command,  the spacing and alignment is not as per the latest User guide ( pg 23 / 24 )

File: docs/Install/HPE_Swarm_Learning_installation.md [hunk 24]
```diff
@@ -2,6 +2,6 @@
 
 1.   [Download the License Server](Download_the_License_Server.md) 
 2.   [Install the License Server and download Swarm Learning](Install_the_License_Server.md) 
-3.   [Install HPE Swarm Learning](Install_HPE_Swarm_Learning.md) 
+3.   [Running SLM-UI Installer](Running_SLM-UI_Installer.md)
```
Comment (suresh-ls) [inline docs/Install/HPE_Swarm_Learning_installation.md]: Not fully done ?  I had given a few lines to be added here in the original comment, right ?

File: docs/User/SWCI_APIs.md (hunk 17 above)
Comment (ganesaar) [reply docs/User/SWCI_APIs.md:11]: Resolved

File: docs/User/SWCI_APIs.md (hunk 18 above)
Comment (ganesaar) [reply docs/User/SWCI_APIs.md]: Resolved

File: lib/src/README.md (hunk 19 above)
Comment (ganesaar) [reply lib/src/README.md]: Resolved

File: docs/User/Swarm_Learning_Command_Interface.md (hunk 23 above)
Comment (ganesaar) [reply docs/User/Swarm_Learning_Command_Interface.md]: Not able to do

File: docs/User/Frequently_asked_questions.md [hunk 25]
```diff
@@ -34,13 +48,13 @@ HPE recommends starting up to 4 SLs to 1 SN and scale it up slowly if needed.
...
 If you are starting the SL and ML nodes by using the `run-sl` script, then the GPUs can be specified as appropriate environment variables by using `--ml-e` option.
 
 For AMD GPUs, you can set `usercontaineropts` and/or `usrenvvars` section of the SWOP profile. For more information, see [SWOP profile schema](/docs/User/SWOP_profile_schema.md).
-If you are starting the SL and ML nodes by using the `run-sl` script, then the GPUs can be specified as appropriate parameters as specified in the [User machine learning container parameters](/docs/Install/Running_Swarm_Learning.md).
+If you are starting the SL and ML nodes by using the `run-sl` script, then the GPUs can be specified as appropriate parameters as specified in the [User machine learning container parameters](/docs/Install/Running_Swarm_Learning.md). For more information on User machine learning container parameters, see *HPE Swarm Learning Installation and Configuration Guide*.
```
Comment (suresh-ls) [inline docs/User/Frequently_asked_questions.md]: Stale reference to install guide..

remove this "For more information on User machine learning container parameters, see *HPE Swarm Learning Installation and Configuration Guide*."

File: docs/User/SWCI_APIs.md [hunk 26]
```diff
@@ -3,17 +3,19 @@
 The following API methods help to invoke SWCI operations programmatically. It is used to view the status, control, and manage the Swarm Learning framework.
 
 <blockquote>
-NOTE:Arguments that do not have default values are mandatory.
+NOTE:
+  Arguments that do not have default values are mandatory.
+  This is an experimental API methods for developers.
```
Comment (suresh-ls) [inline docs/User/SWCI_APIs.md:8]: API methods -> API feature

File: examples/contrib/breakhis/model/breakhis.py (hunk 22 above)
Comment (suresh-ls) [reply examples/contrib/breakhis/model/breakhis.py]: Ignore

**Comment by RadhakrishnaJ** [comment]
Merging 2.0 doc content to master.

**Pull request merged**